atef/config.py

```diff
@@ -193,7 +347,7 @@ def from_device(
         comparison: Comparison,
         name: Optional[str] = None,
         path: Optional[List[PathItem]] = None,
-    ) -> PreparedComparison:
+    ) -> PreparedSignalComparison:
```

Comment:
Slightly off-topic: are these `from_device` and `from_pvname` methods the proper place to get and stash the dynamic values in #111? (and `from_tool` for the off chance someone wants to... compare a ping ms to a PV?)

Reply:
I'd say that's pretty on-topic as your PR's needs should drive the API choices in here as well.

As of now, "prepare" means "take comparison, device, attribute and create the outer product of possible checks".

Given that we want to grab external values once, I'd say those dynamic values probably _should_ be retrieved here (just once, no matter how many times they are used).

It'd be nice to not block on `from_device` while making up all of these prepared comparisons, though. I think this needs some more thought.

Reply:
> It'd be nice to not block on from_device while making up all of these prepared comparisons, though. I think this needs some more thought.

It sounds like this needs some sort of async-light callback that stashes the value when it comes in and flags that the values is ready